models/settings_helpers.go

```diff
@@ -109,6 +113,9 @@ func UpdateSettings(q reform.DBTX, params *ChangeSettingsParams) (*Settings, err
 		settings.AlertManagerURL = ""
 	}
 
+	// because of validation, both variables cannot be true at the same time
+	settings.EnableSTT = params.EnableSTT && !params.DisableSTT
```

Comment:
Let's do it in a more explicit and "dumb" way like for telemetry above. If something needs a comment explaining the logic, then it probably could be simpler. :)